gateway/managers/syncman/syncman_crud.go

```diff
@@ -542,84 +542,23 @@ func (s *Manager) GetCollectionRules(ctx context.Context, project, dbAlias, col
 	return http.StatusOK, []interface{}{coll}, nil
 }
 
-type dbJSONSchemaResponse struct {
-	Fields []*model.FieldType `json:"fields"`
-}
-
 // GetSchemas gets schemas from config
 func (s *Manager) GetSchemas(ctx context.Context, project, dbAlias, col, format string, params model.RequestParams) (int, []interface{}, error) {
 	// Acquire a lock
 	s.lock.Lock()
 	defer s.lock.Unlock()
 
-	projectConfig, err := s.getConfigWithoutLock(ctx, project)
+	_, err := s.getConfigWithoutLock(ctx, project)
 	if err != nil {
 		return http.StatusBadRequest, nil, err
 	}
 
-	if format == "json" {
-		a := s.modules.GetSchemaModuleForSyncMan()
-		if dbAlias != "*" && col != "*" {
-			collectionInfo, p := a.GetSchema(dbAlias, col)
-			if !p {
-				return http.StatusBadRequest, nil, helpers.Logger.LogError(helpers.GetRequestID(ctx), fmt.Sprintf("collection (%s) not present in config for dbAlias (%s) )", dbAlias, col), nil, nil)
-			}
-			fields := []*model.FieldType{}
-			for _, v := range collectionInfo {
-				fields = append(fields, v)
-			}
-			return http.StatusOK, []interface{}{map[string]*dbJSONSchemaResponse{fmt.Sprintf("%s-%s", dbAlias, col): {Fields: fields}}}, nil
-		} else if dbAlias != "*" {
-			collections := projectConfig.Modules.Crud[dbAlias].Collections
-			coll := map[string]*dbJSONSchemaResponse{}
-			for key := range collections {
-				collectionInfo, _ := a.GetSchema(dbAlias, key)
-				fields := []*model.FieldType{}
-				for _, v := range collectionInfo {
-					fields = append(fields, v)
-				}
-				coll[fmt.Sprintf("%s-%s", dbAlias, key)] = &dbJSONSchemaResponse{Fields: fields}
-			}
-			return http.StatusOK, []interface{}{coll}, nil
-		}
-		databases := projectConfig.Modules.Crud
-		coll := map[string]*dbJSONSchemaResponse{}
-		for dbName, dbInfo := range databases {
-			for key := range dbInfo.Collections {
-				collectionInfo, _ := a.GetSchema(dbName, key)
-				fields := []*model.FieldType{}
-				for _, v := range collectionInfo {
-					fields = append(fields, v)
-				}
-				coll[fmt.Sprintf("%s-%s", dbName, key)] = &dbJSONSchemaResponse{Fields: fields}
-			}
-		}
-		return http.StatusOK, []interface{}{coll}, nil
+	arr, err := s.modules.GetSchemaModuleForSyncMan().GetSchemaForDB(ctx, dbAlias, col, format)
```

Comment:
Get the schema module and call the `GetSchemaForDB` in two different steps. It will cause a problem otherwise in enterprise version